common/src/main/java/com/wynntils/features/inventory/CustomBankPagesFeature.java

```diff
@@ -0,0 +1,221 @@
+                    buttonThreePage.get(),
+                    buttonFourPage.get(),
+                    buttonFivePage.get(),
+                    buttonSixPage.get());
+
+            if (currentPage != pageDestination && quickJumping) {
+                goToPage();
+            } else if (quickJumping) {
+                quickJumping = false;
+            }
+        } else {
+            return;
+        }
+    }
+
+    @SubscribeEvent
+    public void onContainerClose(ContainerCloseEvent.Post e) {
+        currentPage = 1;
+        pageDestination = 1;
+    }
+
+    @SubscribeEvent
+    public void onSlotClicked(ContainerClickEvent e) {
+        if (e.getContainerMenu().containerId != containerId) {
+            return;
+        }
+
+        int slotIndex = e.getSlotNum();
+
+        if (BUTTON_SLOTS.contains(slotIndex)) {
+            int buttonIndex = BUTTON_SLOTS.indexOf(slotIndex);
+            pageDestination = Math.max(1, Math.min(MAX_BANK_PAGES, customJumpDestinations.get(buttonIndex)));
+            e.setCanceled(true);
+            goToPage();
+        } else if (slotIndex == NEXT_PAGE_SLOT) {
+            pageDestination++;
+        } else if (slotIndex == PREVIOUS_PAGE_SLOT) {
+            pageDestination--;
+        }
+    }
+
+    @SubscribeEvent
+    public void onRenderSlot(SlotRenderEvent.Pre e) {
+        if (!(e.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
+
+        StyledText title = StyledText.fromComponent(screen.getTitle());
+
+        Matcher pageMatcher = title.getMatcher(BANK_PAGE_MATCHER);
+
+        if (pageMatcher.matches()) {
+            if (BUTTON_SLOTS.contains(e.getSlot().index)) {
+                renderQuickJumpButton(e.getPoseStack(), e.getSlot());
+            }
+        }
+
+        // Prevent buying a new page
+        if (e.getSlot().index == NEXT_PAGE_SLOT
+                && e.getSlot().getItem().getHoverName().getString().contains("§c")
+                && pageDestination > currentPage) {
+            pageDestination = currentPage;
+            quickJumping = false;
+        }
+    }
+
+    private void renderQuickJumpButton(PoseStack poseStack, Slot buttonSlot) {
+        RenderUtils.drawTexturedRect(
+                poseStack,
+                Texture.COLUMN_ARROW_RIGHT.resource(),
+                buttonSlot.x - 8,
+                buttonSlot.y - 8,
+                300,
+                32,
+                32,
+                Texture.COLUMN_ARROW_RIGHT.width(),
+                Texture.COLUMN_ARROW_RIGHT.height());
+
+        buttonSlot.set(new ItemStack(Items.SNOW));
+
+        int buttonIndex = BUTTON_SLOTS.indexOf(buttonSlot.index);
+        // When config limits 1-21, change to just customJumpDestinations.get(buttonIndex)
+        int buttonDestination = Math.max(1, Math.min(MAX_BANK_PAGES, customJumpDestinations.get(buttonIndex)));
+
+        buttonSlot.getItem().setHoverName(Component.literal("§7Jump to Page " + buttonDestination));
```

Comment:
Can you use the ChatFormatting enum instead of raw § codes?

Reply:
I think that is fine. We do that elsewhere, I think. @kristofbolyai, what do you say?

Reply:
We use them in a few places yes, while it is usually recommended to use ChatFormatting, it is not strictly necessary.